scrapers/ar/bills.py

```diff
@@ -232,32 +239,57 @@ def scrape_bill_page(self, bill):
         page = lxml.html.fromstring(page)
         page.make_links_absolute(url)
 
+        primary_sponsors_path = page.xpath(
+            "//div[text()[contains(.,'Lead Sponsor:')]]/../div[2]/a"
+        )
+        for a in primary_sponsors_path:
+            primary_sponsors = a.text_content().strip()
+            primary_sponsors_link = a.attrib["href"]
+            chamber = self.scrape_chamber(bill, primary_sponsors_link)
+
+            bill.add_sponsorship(
+                primary_sponsors,
+                classification="primary",
+                entity_type=self.get_entity_name(primary_sponsors_link),
+                primary=True,
+                chamber=chamber,
+            )
+
         other_primary_sponsors_path = page.xpath(
             "//div[text()[contains(.,'Other Primary Sponsor:')]]/../div[2]/a"
         )
         for a in other_primary_sponsors_path:
             other_primary_sponsors = a.text_content().strip()
+            other_primary_sponsors_link = a.attrib["href"]
+            chamber = self.scrape_chamber(bill, other_primary_sponsors_link)
+
             bill.add_sponsorship(
                 other_primary_sponsors,
                 classification="primary",
-                entity_type="person",
+                entity_type=self.get_entity_name(other_primary_sponsors_link),
                 primary=True,
+                chamber=chamber,
             )
 
         cosponsor_path = page.xpath(
             "//div[text()[contains(.,'CoSponsors:')]]/../div[2]/a"
         )
         for a in cosponsor_path:
             cosponsor = a.text_content().strip()
+            cosponsor_link = a.attrib["href"]
+            chamber = self.scrape_chamber(bill, cosponsor_link)
+
             bill.add_sponsorship(
                 cosponsor,
                 classification="cosponsor",
-                entity_type="person",
+                entity_type=self.get_entity_name(cosponsor_link),
                 primary=False,
+                chamber=chamber,
             )
 
         try:
             cosponsor_link = page.xpath("//a[contains(@href, 'CoSponsors')]")[0]
+
             self.scrape_cosponsors(bill, cosponsor_link.attrib["href"])
```

Comment:
I notice there is still a call here to `self.scrape_cosponsors()` here, even though the definition of the function seems to be removed below (old line 455, replaced by new line 487 `scrape_chamber()`) - is there another definition of  `self.scrape_cosponsors()` being inherited, or will those throw an error if this code path is reached?

Reply:
We don't need that code anymore. I've removed this code.